### ISSUE TITLE
Proposal: ParentNode.wrap()

### ISSUE THREAD

**Issue by jonathantneal** [whatwg/dom#479]
The **ParentNode.wrap()** method would replace a node with the parent node, appending the replaced node to the parent node.

### Syntax

```js
ParentNode.wrap(node);
```

### Rationale:

Like `childNode.remove()`, we’re de-complicating the process of wrapping a node, e.g.:

```js
if (node.parentNode) {
  node.parentNode.insertBefore(parentNode, node).appendChild(node);
}
```

**Comment by domenic** [comment]
The "Syntax" section doesn't make sense, as `ParentNode` is not an exposed global, so creating a static method on it doesn't work. Could you give an example?

**Comment by jonathantneal** [comment]
@domenic, you are right. I reversed how this is done in existing libraries, and even my own production code. While writing this proposal, I fooled myself into reversing them. Please let me know if things make more sense now.

**Comment by snuggs** [comment]
@jonathantneal @domenic FWIW I believe the implementation of this can lean upon the `Document.execCommand ('formatBlock', undefined, wrapWithThisTag)` API. The only caveat is `formatBlock` uses selected content but conceptually similar in implementation. https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

**Comment by travisleithead** [comment]
For historical context, a semantically [similar API](https://msdn.microsoft.com/en-us/library/ff520928(v=vs.85).aspx) existed in the earliest versions of Internet Explorer as `elem.applyElement(child, "outside")`. The platform has started a trend toward 'this' operations, like (`remove`, `replaceWith`), so this idea is not entirely out-of-place anymore.
